src/main/java/org/jenkinsci/plugins/pipeline/github/client/Review.java

```diff
@@ -0,0 +1,66 @@
+package org.jenkinsci.plugins.pipeline.github.client;
+
+import java.io.Serializable;
+import java.text.MessageFormat;
+
+import org.eclipse.egit.github.core.User;
+
+public class Review implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    public static final String STATE_APPROVED = "APPROVED";
+    public static final String STATE_PENDING = "PENDING";
+    public static final String STATE_CHANGES_REQUESTED = "CHANGES_REQUESTED";
+    public static final String STATE_DISMISSED = "DISMISSED";
+    public static final String STATE_COMMENTED  = "COMMENTED";
+
+    private String body;
+    private long id;
+    private User user;
+    private String state;
+
+    public String getBody() {
+        return body;
+    }
+
+    public Review setBody(final String body) {
+        this.body = body;
+        return this;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public Review setId(final long id) {
+        this.id = id;
+        return this;
+    }
+
+    public User getUser() {
+        return user;
+    }
+
+    public Review setUser(final User user) {
+        this.user = user;
+        return this;
+    }
+
+    public String getState() {
+        return state;
+    }
+
+    /**
+     * @param state one of APPROVED, PENDING, CHANGES_REQUESTED, DISMISSED, COMMENTED
+     * @return this review
+     * throws {@link IllegalArgumentException} if state is invalid
+     */
+    public Review setState(final String state) {
+        if (STATE_APPROVED.equals(state) || STATE_PENDING.equals(state) || STATE_CHANGES_REQUESTED.equals(state)
```

Comment:
You could put the valid states in a Set<> in a static {} block, then check if the provided state is in the Set.

Reply:
Used a set in https://github.com/jenkinsci/pipeline-github-plugin/pull/18/commits/a84d222694a33f13995ffeb6c490e0270d371c9f.

This was based on [similar implementations in the egit-github library](https://github.com/eclipse/egit-github/blob/v4.11.0.201803080745-r/org.eclipse.egit.github.core/src/org/eclipse/egit/github/core/CommitStatus.java#L151-L163). I stuck with using just String instead of an enum, because of the reasons you mentioned, plus I don't know what the implications are for serialization when using Enums here.